test/diff_web/controllers/page_controller_test.exs

```diff
@@ -1,8 +1,29 @@
 defmodule DiffWeb.PageControllerTest do
   use DiffWeb.ConnCase
+  import Mox
 
   test "GET /", %{conn: conn} do
     conn = get(conn, "/")
     assert html_response(conn, 200) =~ "<!DOCTYPE html>"
   end
+
+  describe "GET /diff/:package/:versions" do
+    setup :verify_on_exit!
+
+    test "does not accept implicit from", %{conn: conn} do
+      conn = get(conn, "/diff/phoenix/%20..")
+      assert html_response(conn, 400) =~ "Bad request"
+    end
+
+    test "accepts implicit to", %{conn: conn} do
+      versions = ["1.4.5", "1.4.9", "1.5.0-rc.2"]
+      diff = "p403n1xd1ff"
+
+      expect(Diff.Package.StoreMock, :get_versions, fn "phoenix" -> {:ok, versions} end)
+      expect(Diff.StorageMock, :get, fn "phoenix", "1.4.5", "1.4.9" -> {:ok, diff} end)
+
+      conn = get(conn, "diff/phoenix/1.4.5..")
```

Comment:
actually, I think this URL is a bit awkward. My other ideas would be:

1. `diff/phoenix/1.4.5`
2. `diff/phoenix/1.4.5..latest`

I think I like option 1. the most. What do you all think?

Reply:
Hey! I'm fine with the url with the `..` hehe, but I understand the awkwardness.

Between 1 and 2 I prefer 2, since the idea is to be clear, 2 seems to me the clearest of them all.

Reply:
Will go that path (2), and squash all the things, in 3, 2, 1… 🙂

Reply:
before making any changes let's wait for other opinions!

Reply:
btw, no worries about squashing commits, we almost always squash when merging.

Reply:
I think `diff/phoenix/1.4.5` looks best.

Reply:
thanks for the input! should we go with `diff/phoenix/1.4.5` then? 🙂

Reply:
https://github.com/hexpm/diff/pull/45/commits/b7127a5adf583af615abf18523dd4cde7af725f9 should do it 👍

good thing that `diff/phoenix/1.4.5..` incidentally also work 😄

Reply:
I think `diff/phoenix/1.4.5..` should be a 400 bad request but I don't have a strong opinion about it.